Makefile

```
@@ -143,10 +148,82 @@ generate: $(TOOLS_DIR)/stringer
 
 .PHONY: license-check
 license-check:
-	@licRes=$$(for f in $$(find . -type f \( -iname '*.go' -o -iname '*.sh' \) ! -path './vendor/*') ; do \
+	@licRes=$$(for f in $$(find . -type f \( -iname '*.go' -o -iname '*.sh' \) ! -path './vendor/*' ! -path './internal/opentelemetry-proto/*') ; do \
 	           awk '/Copyright The OpenTelemetry Authors|generated|GENERATED/ && NR<=3 { found=1; next } END { if (!found) print FILENAME }' $$f; \
 	   done); \
 	   if [ -n "$${licRes}" ]; then \
 	           echo "license header checking failed:"; echo "$${licRes}"; \
 	           exit 1; \
 	   fi
+
+# Find all .proto files.
+OTEL_PROTO_SUBMODULE := internal/opentelemetry-proto
+PROTO_GEN_DIR        := internal/opentelemetry-proto-gen
+PROTOBUF_TEMP_DIR    := gen/pb-go
+PROTOBUF_SOURCE_DIR  := gen/proto
+ORIG_PROTO_FILES     := $(wildcard $(OTEL_PROTO_SUBMODULE)/opentelemetry/proto/*/v1/*.proto \
+                           $(OTEL_PROTO_SUBMODULE)/opentelemetry/proto/collector/*/v1/*.proto)
+SOURCE_PROTO_FILES   := $(subst $(OTEL_PROTO_SUBMODULE),$(PROTOBUF_SOURCE_DIR),$(ORIG_PROTO_FILES))
+
+ifeq ($(CIRCLECI),true)
+copy-protos-to:
+	# create a dummy container to hold the protobufs
+	docker create -v /defs --name proto-orig alpine:3.4 /bin/true
+	# copy from here into the proto-src directory
+	docker cp ./gen proto-orig:/defs
+
+copy-protobuf-from:
+	rm -fr ./$(PROTO_GEN_DIR)
+	# copy from dummy volume back to our source directory
+	docker cp proto-orig:/defs/$(PROTOBUF_TEMP_DIR)/go.opentelemetry.io/otel/$(PROTO_GEN_DIR) ./$(PROTO_GEN_DIR)
+
+else
+copy-protos-to:
+	@:    # nop
+
+copy-protobuf-from:
+	rm -fr ./$(PROTO_GEN_DIR)
+	mv ./$(PROTOBUF_TEMP_DIR)/go.opentelemetry.io/otel/$(PROTO_GEN_DIR) ./$(PROTO_GEN_DIR)
+endif
+
+
+# This step can be omitted assuming go_package changes are made in opentelemetry-proto repo
+define exec-replace-pkgname
+sed  's+go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go+go_package = "go.opentelemetry.io/otel/internal/opentelemetry-proto-gen+' < $(1) > $(2)
+
+endef
+
+.PHONY: protobuf make-protobufs
+protobuf: protobuf-source copy-protos-to make-protobufs copy-protobuf-from
+
+.PHONY: protobuf-source
+protobuf-source: $(SOURCE_PROTO_FILES) | $(PROTOBUF_SOURCE_DIR)/
+
+# replace opentelemetry-proto v0.4.0 package name by repo-local version
+$(SOURCE_PROTO_FILES): $(PROTOBUF_SOURCE_DIR)/%.proto: $(OTEL_PROTO_SUBMODULE)/%.proto
+	@mkdir -p $(@D)
+	$(call exec-replace-pkgname,$<,$@)
+
+ifeq ($(CIRCLECI),true)
+VOLUMES_MOUNT=--volumes-from proto-orig
+else
+VOLUMES_MOUNT=-v `pwd`:/defs
+endif
+
+define exec-protoc-all
+docker run $(VOLUMES_MOUNT) namely/protoc-all $(1)
```

This container needs to be versioned to a tag so when latest is updated it doesn't break things.

E.g.

```suggestion
docker run $(VOLUMES_MOUNT) namely/protoc-all:1.28_2 $(1)
```

Though, it probably makes sense to keep the version as a variable.

---

Perhaps better, pin the image to a digest:

`namely/protoc-all@sha256:b3eaf5e5ed4f3d0848ba5f5eda16374eb1d73e1fdbf30b0f13d439d96da5b81e`